### DISCUSSION TITLE
Table (v8) vs TableInstance (v7)

### DISCUSSION THREAD

**Discussion by petegriffith** [TanStack/table#4116] [Q&A]
In v7, `useTable` returns a `TableInstance` while in v8, `useReactTable` returns a `Table`. I understand the migration guide isn't ready yet, but is there a simple explanation available about how to think about one vs the other? Are they materially different objects?

One difference is that `useTable().headerGroups` (v7) returns a different list of headers than useReactTable().useHeaderGroups() (v8). In my current migration, I'm struggling to figure out how to get v8 to return a list similar to what v7 returned.

**Comment by nikikaranikola** [comment]
Yes, I'm finding the same problem and don't know how to solve it! Did you figure it out?

**Reply by FullstackWEB-developer** [reply]
I am struggling with the same problem.
Please share solution.
